server/src/api/v1.py

```diff
@@ -489,6 +493,48 @@ def agents_post(agent_name, json_data):
     return "OK"
 
 
+@v1.post("/agents/status")
```

Comment:
After our chat today I was looking at how you would add the resource_url and realized that you don't have the job_id here anywhere. Then I got to thinking more about the path for this API and I'm not sure if `/v1/agents/status` is the right place (also the README is wrong but I'll comment on that separately). I think putting it in this path would hint that it's a status update about "agents in general". In the case of this API though, it's that at all. Instead, we're sending events that have happened in the context of a **job**. Sure, it has a specific agent associated with it, and a queue, but it's all data about what happened during the job itself. Also, associating it with a job_id means we can send that information across quite easily.

What do you think about putting it instead, at: `/job/<job_id>/events` or something like that? I'm open to other ideas, but I feel like this is at least a bit more consistent with what we already have, and conveys the meaning a bit better. Also, it's a pretty simple change, and it gets you the job_id that you're also going to want to insert as the `resource_url`

Reply:
Does the server need the job_id? If we're going to put the resource_url in the JOB_START event, it sounds easier to have the agent create the url with the job_id than have the server parse the events and create the url for the JOB_START event.

Reply:
@val500 the main thing I was looking for here was a small change to the endpoint which it looks like you took care of. However, I don't see where that resource_url is actually injected?

Reply:
It's in the detail for the JOB_START event

Reply:
I think the URL is actually incorrect here - would it be {server_name}/job/{job_id}?

Reply:
right now, it looks like you have `f"{self.client.server}/job/{job.job_id}/events"` but I think this is wrong, I was under the impression that this should be a link to the url for the job details page, which would be something like `f"{self.client.server}/jobs/{job.job_id}"` (note the "s" in "jobs" also). Let's sync up about this tomorrow as I think Omar also had some related comments on the TO side.

Reply:
Yes we'd need the job link similar to https://testflinger.canonical.com/jobs/697b8930-3d88-48d0-8458-aafe2820e312 for example